spec.bs

```diff
@@ -440,3 +461,17 @@ All call sites of HTML's existing <a spec="html">resolve a module specifier</a>
 * [=Fetch an import() module script graph=] will also need to take a [=script=] instead of a base URL.
 
 Call sites will also need to be updated to account for [=resolve a module specifier=] now throwing exceptions, instead of returning failure. (Previously most call sites just turned failures into {{TypeError}}s manually, so this is straightforward.)
+
+In addition to the call sites for [=validate a module script URL=] explicitly added within this spec, insert the following:
+
+<div algorithm="check-access-snippet">
+  1. [=Validate the module script URL=] given <var ignore>url</var>, |settings object|, and |settings object|'s [=environment settings object/API base URL=]. If this throws an error, then asynchronously complete this algorithm with null, and abort these steps.
+</div>
+
+at the beginning of the following HTML spec concepts (after [=wait for import maps=]):
+
+- [=fetch an external module script graph=]
+- [=fetch a modulepreload module script graph=]
+- [=fetch a module worker script graph=] (using <var ignore>module map settings object</var>)
```

Comment:
It seems it might be cleaner to add the check at the beginning of "fetch a single module script". That way if we ever add new entry points (e.g. worklets?) those are covered automatically.

That would lead to redundant checks though, so engines probably wouldn't want to follow the spec exactly, which is annoying... I guess maybe the goal of easily adding new entry points is not going to work out. (E.g. they also need to wait for import maps.)

Reply:
Yeah we can do this, if we allow redundant checks (i.e. calling validation twice for every non-toplevel fetch) as you mentioned. I'm a little concerned about the double checking pattern, because if a new non-const operation follows the same pattern in the future (e.g. it is added to the validation, or added to the spec in a similar fashion), it would be broken. (also I feel there was a similar bug due to doing something twice around dynamic import)

I switched to "fetch a single module script" approach, but still leave the original approach in a note.